tests/run-make/c-link-to-rust-dylib/rmake.rs

```diff
@@ -0,0 +1,44 @@
+// This test checks that C linking with Rust does not encounter any errors, with dynamic libraries.
+// See <https://github.com/rust-lang/rust/issues/10434>.
+
+//@ ignore-cross-compile
+
+use std::fs::remove_file;
+
+use run_make_support::{
+    cc, dynamic_lib_extension, is_msvc, read_dir, run, run_fail, rustc, tmp_dir,
+};
+
+fn main() {
+    rustc().input("foo.rs").run();
+
+    if is_msvc() {
+        let lib = tmp_dir().join("foo.dll.lib");
+
+        // We remove the file in case it exists.
+        let _ = remove_file(&lib);
+        cc().input("bar.c").arg(lib).out_exe("bar").run();
+    } else {
+        cc().input("bar.c")
+            .arg("-lfoo")
+            .output(tmp_dir().join("bar"))
+            .library_search_path(tmp_dir())
+            .run();
+    }
+
+    run("bar");
+
+    let expected_extension = dynamic_lib_extension();
+    read_dir(tmp_dir(), |path| {
+        if path.is_file()
+            && path.extension().is_some_and(|ext| ext == expected_extension)
+            && path
+                .file_name()
+                .and_then(|name| name.to_str())
+                .is_some_and(|name| name.starts_with("lib"))
```

Comment:
This condition is not quite right:

```rs
// from tools.mk

ifeq ($(UNAME),Darwin)
DYLIB_GLOB = lib$(1)*.dylib
else
ifdef IS_WINDOWS
DYLIB_GLOB = $(1)*.dll
ifdef IS_MSVC
else
endif
else
DYLIB_GLOB = lib$(1)*.so
endif
endif
```

the `lib` prefix doesn't exist when it's `.dll`.

Reply:
We also have the `foo.dll.lib` file, the removal should probably also be near here.

Reply:
Good catch! Gonna remove all `.so`, `.dll` and `.dylib`, will be simpler.